packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -375,22 +454,78 @@ export class ChildLogger extends TelemetryLogger {
 		protected readonly baseLogger: ITelemetryBaseLogger,
 		namespace: string | undefined,
 		properties: ITelemetryLoggerPropertyBags | undefined,
+		private readonly eventSamplingConfigs?: Map<string, SamplingConfig>,
 	) {
 		super(namespace, properties);
 
 		// propagate the monitoring context
 		if (loggerIsMonitoringContext(baseLogger)) {
 			mixinMonitoringContext(this, new CachedConfigProvider(this, baseLogger.config));
+
+			// Read config flag only once, so we don't pay the price every time an event is logged.
+			const mc = loggerToMonitoringContext(this);
+			if (mc.config.getBoolean("Fluid.Telemetry.DisableSampling") === true) {
+				this.isSamplingDisabled = true;
+			}
 		}
 	}
 
 	/**
-	 * Send an event with the logger
+	 * Send an event with the logger. If the given event name is mapped to a {@link SamplingConfig}
```

Comment:
Nit (for a minor generated docs improvement): 
```suggestion
	 * Send an event with the logger.
	 * 
	 * If the given event name is mapped to a {@link SamplingConfig}
```